src/main/java/io/gingersnapproject/cdc/remote/RemoteOffsetStore.java

```diff
@@ -6,23 +6,25 @@
 import java.util.Map;
 import java.util.concurrent.Future;
 
-import org.apache.kafka.connect.runtime.WorkerConfig;
-import org.apache.kafka.connect.storage.OffsetBackingStore;
-import org.apache.kafka.connect.util.Callback;
 import io.gingersnapproject.cdc.OffsetBackend;
 import io.gingersnapproject.cdc.cache.CacheService;
-import io.gingersnapproject.cdc.cache.ErrorNotifier;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import io.gingersnapproject.cdc.event.InternalEventManager;
+import io.gingersnapproject.util.ArcUtil;
 
 import io.quarkus.arc.Arc;
 import io.quarkus.arc.InstanceHandle;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.OffsetBackingStore;
+import org.apache.kafka.connect.util.Callback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class RemoteOffsetStore implements OffsetBackingStore {
    public static final String URI_CACHE = "offset.storage.remote.uri";
    public static final String TOPIC_NAME = "offset.storage.remote.topic";
    private static final Logger log = LoggerFactory.getLogger(RemoteOffsetStore.class);
 
+   private InternalEventManager eventing;
```

Comment:
Why can't we inject this to avoid using Arc explicitly?

Reply:
This one and the `RemoteSchemaHistory` are instantiated internally by Debezium. We could try to insert the `InternalEventManager` into the configuration object and these two classes could acquire it there, without relying on Arc. WDYT?

Reply:
I added the `@Inject` annotation to `RemoteOffsetStore` and `RemoteSchemaHistory` and it seems to work.

Reply:
Very nice, I'll try that

Reply:
I tried using the `@Inject` for both offset and schemas stores, but the value remains null.

Reply:
Argh my bad, I didn't explicitly check the value. I just did an e2e test and obviously didn't hit that code.